src/Client/ClientBase.h

```diff
@@ -218,6 +218,24 @@ class ClientBase : public Poco::Util::Application
     int query_fuzzer_runs = 0;
 
     QueryProcessingStage::Enum query_processing_stage;
+
+    struct HostPort
+    {
+        String host;
+        std::optional<int> port{};
+        friend std::istream & operator>>(std::istream & in, HostPort & hostPort) {
+            String
+                host_with_port,
+                delimiter = ":";
+            in >> host_with_port;
+            size_t delimiter_pos = host_with_port.find(delimiter);
+            hostPort.host = host_with_port.substr(0, delimiter_pos);
```

Comment:
You can move it also inside `if` and separately assign `hostPort.host` to `host_with_port` in `else` branch so that not to mess with `host_with_port.substr(0, std::string::npos)`

Reply:
Done